docs/concepts/ipns.md

```diff
@@ -5,19 +5,177 @@ description: Learn about the InterPlanetary Name System (IPNS) and how it can be
 
 # InterPlanetary Name System (IPNS)
 
-IPFS uses [content-based addressing](content-addressing.md); it creates an address of a file based on data contained within the file. If you were to share an IPFS address such as `/ipfs/QmbezGequPwcsWo8UL4wDF6a8hYwM1hmbzYv2mnKkEWaUp` with someone, you would need to give the person a new link every time you update the content.
+- [InterPlanetary Name System (IPNS)](#interplanetary-name-system-ipns)
+  - [Mutability in IPFS](#mutability-in-ipfs)
+  - [How IPNS works](#how-ipns-works)
+    - [Anatomy of an IPNS name](#anatomy-of-an-ipns-name)
+      - [How IPNS names relate to CIDs](#how-ipns-names-relate-to-cids)
+    - [IPNS names are self-certifying](#ipns-names-are-self-certifying)
+    - [Common IPNS operations](#common-ipns-operations)
+    - [IPNS is transport agnostic](#ipns-is-transport-agnostic)
+      - [IPNS over the DHT](#ipns-over-the-dht)
+      - [IPNS over PubSub](#ipns-over-pubsub)
+        - [Publishing IPNS records over PubSub lifecycle](#publishing-ipns-records-over-pubsub-lifecycle)
+    - [Tradeoffs between consistency vs. availability](#tradeoffs-between-consistency-vs-availability)
+      - [IPNS record validity](#ipns-record-validity)
+      - [Practical considerations](#practical-considerations)
+  - [IPNS in practice](#ipns-in-practice)
+    - [Resolving IPNS names using IPFS gateways](#resolving-ipns-names-using-ipfs-gateways)
+  - [Publishing IPNS names with Kubo](#publishing-ipns-names-with-kubo)
+  - [Example IPNS Setup with JS SDK API](#example-ipns-setup-with-js-sdk-api)
+  - [Alternatives to IPNS](#alternatives-to-ipns)
+  - [Further Resources](#further-resources)
 
-The InterPlanetary Name System (IPNS) solves this issue by creating an address that can be updated.
 
-A _name_ in IPNS is the [hash](hashing.md) of a public key. It is associated with a record containing information about the hash it links to that is signed by the corresponding private key. New records can be signed and published at any time.
+## Mutability in IPFS
 
-When looking up an IPNS address, use the `/ipns/` prefix:
+[Content addressing](content-addressing.md) in IPFS is by nature _immutable_: when you add a file to IPFS, it creates a hash from the data contained in it. Changing a file changes its hash, and consequently its CID which is used as an address.
+
+Yet, there are many situations where **mutable pointers** are useful as a complement to immutability, for example, when publishing a website that frequently changes. It would be impractical to share a new CID every time you update the website. With mutable pointers, you can share the address of the pointer once, and update the pointer – to the new CID – every time you publish a change.
```

Comment:
```suggestion
Yet, there are many situations where content-addressed data needs to be regularly updated, for example, when publishing a website that frequently changes. It would be impractical to share a new CID every time you update the website. With **mutable pointers**, you can share the address of the pointer once, and update the pointer – to the new CID – every time you publish a change.
```